crop_classification/dl/timl.py

```diff
@@ -717,3 +724,42 @@ def train_timl_model(
     )
 
     return model.model
+
+
+def load_timl_model(
+    task_info: np.ndarray,
+    input_size: int,
+    num_timesteps: int,
+    model_state_dict_path: Path,
+    encoder_state_dict_path: Optional[Path],
+):
+    """
+    Load a trained TIML model
+    """
+    model = Classifier(
+        input_size=input_size,
+        classifier_base_layers=CLASSIFIER_BASE_LAYERS,
+        num_classification_layers=NUM_CLASSIFICATION_LAYERS,
+        classifier_dropout=CLASSIFIER_DROPOUT,
+        classifier_vector_size=HIDDEN_VECTOR_SIZE,
+    )
+    model.load_state_dict(torch.load(model_state_dict_path))
+
+    if encoder_state_dict_path.exists():
+        encoder = TaskEncoder(
+            input_size=task_info.shape[0],
+            encoder_vector_sizes=ENCODER_VECTOR_SIZES,
+            encoder_dropout=ENCODER_DROPOUT,
+            num_bands=input_size,  # TODO is this right?
```

Comment:
This was `num_bands=dataset.num_bands` before but I believe if the encoder is enabled then `input_size` is `dataset.num_bands`. Let me know if ok

Reply:
this works :)